assets/js/base/hocs/with-window.js

```diff
@@ -0,0 +1,8 @@
+const w = window || {};
+
+const withWindow = ( propMap ) => ( WrappedComponent ) => ( incomingProps ) => {
+	const props = propMap( w, incomingProps );
+	return <WrappedComponent { ...incomingProps } { ...props } />;
+};
```

Comment:
Right, I was thinking that it might cause unnecessary re-renders if `incomingProps` doesn't change because `propMap` returns a new object.  However, I think we're okay because we're spreading the objects as props on the wrapped component.